pkg/discovery/backends.go

```diff
@@ -39,6 +40,70 @@ func parse(rawurl string) (string, string) {
 	return parts[0], parts[1]
 }
 
+// ParseAdvertise parses the --cluster-advertise daemon config which accepts
+// <ip-address>:<port> or <interface-name>:<port>
+func ParseAdvertise(store, advertise string) (string, error) {
+	var (
+		iface *net.Interface
+		addrs []net.Addr
+		err   error
+	)
+
+	addrPort := strings.Split(advertise, ":")
```

Comment:
Good point. But we dont support ipv6 as well for advertise. having said that, it is more appropriate to use SplitHostPort and i will push an update.